source/wp-content/themes/wporg-developer-2023/inc/autocomplete.php

```diff
@@ -108,13 +108,9 @@ public function autocomplete_data_update() {
 			wp_send_json_error( $defaults );
 		}
 
-		foreach ( $form_data['post_type'] as $key => $post_type ) {
-			if ( ! in_array( $post_type, $parser_post_types ) ) {
-				unset( $form_data['post_type'][ $key ] );
-			}
-		}
-
-		$post_types = ! empty( $form_data['post_type'] ) ? $form_data['post_type'] : $parser_post_types;
```

Comment:
Removing this logic is a bit concerning. It looks like line 117 was intended to do what we want... is `$form_data['post_type']` not `command` like expected? Or is it stripped by line 113 because it's not in `$parser_post_types`?

Reply:
The code here is a bit tricky and confusing.
1. Originally it used  `$form_data['post_type']`, but according to [this line](https://github.com/WordPress/wporg-developer/pull/351/files/b0a77cfda6d111fcd46895648109cfa476f137c2#diff-9133227a1757e0f7ff2996d7bc952b9db0445049a6984bbd91830d7c6107e43aL104) and [this line](https://github.com/WordPress/wporg-developer/pull/351/files/b0a77cfda6d111fcd46895648109cfa476f137c2#diff-5e0d1972031d52467147da1a95b1eaf3929b263e66640ad2b369130ab4789ef5R97),  `$form_data` seems always `$parser_post_types` (coming from [L94](https://github.com/WordPress/wporg-developer/pull/351/files/b0a77cfda6d111fcd46895648109cfa476f137c2#diff-9133227a1757e0f7ff2996d7bc952b9db0445049a6984bbd91830d7c6107e43aL94))
2. So I was thinking to add `command` to `get_parsed_post_types()`, but it would affect quite a few other parts of the code base.
3. and since `$form_data['post_type']` is always `$parser_post_types` (or is there a case that a user needs to pass in the `post_type` through a form? I don't think this is common for search bar), this check seems extra and `$post_types` in line 117 can just be set as `$parser_post_types` (`post_type` in [defaults](https://github.com/WordPress/wporg-developer/pull/351/files/b0a77cfda6d111fcd46895648109cfa476f137c2#diff-9133227a1757e0f7ff2996d7bc952b9db0445049a6984bbd91830d7c6107e43aL94) is also removed accordingly)
4. Then, because of 2., I didn't add `command` directly to that function, I created an additional condition check for it. ([this](https://github.com/WordPress/wporg-developer/pull/351/files/b0a77cfda6d111fcd46895648109cfa476f137c2#diff-5e0d1972031d52467147da1a95b1eaf3929b263e66640ad2b369130ab4789ef5R99) and [this](https://github.com/WordPress/wporg-developer/pull/351/files/b0a77cfda6d111fcd46895648109cfa476f137c2#diff-9133227a1757e0f7ff2996d7bc952b9db0445049a6984bbd91830d7c6107e43aR111))
5. And yes, stripped by line 113 is one of the reasons.

Overall, my idea is that `$form_data['post_type']` seems always `$parser_post_types` and it doesn't make much sense to get the `$post_type` [from](https://github.com/WordPress/wporg-developer/pull/351/files/b0a77cfda6d111fcd46895648109cfa476f137c2#diff-9133227a1757e0f7ff2996d7bc952b9db0445049a6984bbd91830d7c6107e43aL103-L104) `$_POST['data']`.

Reply:
ps. you might need to expand the code sections to view the reference lines for some of the links.

Reply:
OK I've stepped through the code now and it makes sense to me. It does make me wonder if we can extend the autocomplete to all the other handbooks in the same manner (`/themes`, `/rest-api`, etc). That can be a separate issue though.

Reply:
Ticket for it was created. https://github.com/WordPress/wporg-developer/issues/367